### PR TITLE
Add pre-commit hooks.  Closes #78.

### LINKED ISSUE #78: Add pre-commit/pre-commit to yelp-api
https://github.com/pre-commit/pre-commit has nice linters and checks across several languages.  Sounds like a nice thing to have here.

### PR DESCRIPTION
**Pull request by mittonk**
Sets up configuration for https://github.com/pre-commit/pre-commit and gets our repo to pass those checks, but does _not_ install pre-commit (since there's no python virtualenv / gem list / whatever).

People with Python and pre-commit will get source code checking; those without Python or pre-commit shouldn't notice any difference.

### CODE REVIEW

**Comment by mittonk** [comment]
@dlgallagher , @tomelm @asottile : How does this look for a multi-language-friendly pre-commit install?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-   repo: git://github.com/pre-commit/pre-commit-hooks
+    sha: 29bf11d13689a0a9a895c41eb3591c7e942d377d
+    hooks:
+    -   id: autopep8-wrapper
+    -   id: check-json
+    -   id: check-yaml
+    -   id: check-xml
+    -   id: debug-statements
+    -   id: flake8
+    -   id: trailing-whitespace
```
Comment (asottile) [inline .pre-commit-config.yaml]: probably also want end-of-file-fixer and perhaps a couple more listed here: http://pre-commit.com/hooks.html

**Comment by asottile** [comment]
fwiw according to buck python is installed more often than bash so "not having python" in general isn't a problem.

That said, at the least you can include them in CI to keep your stuff working well.  Those that install them during development'll get faster feedback.

pre-commit aims to be useful to any language, though hooks haven't necessarily been written for all of them yet :D

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mittonk) [inline .pre-commit-config.yaml]: Added end-of-file-fixer, that looks like a good first set.  I'm really reaching for PHP, Perl, and C# hooks at this point, that's where most of the repo's code is.

**Pull request merged**